utils/mil_ros_tools/online_bagger/nodes/online_bagger.py

```diff
@@ -0,0 +1,427 @@
+            rospy.logdebug('number of topic messages: %s', self.get_topic_message_count(topic))
+
+    def get_subscribed_topics(self):
+
+        """
+        Return subscribed and failed topic list
+        """
+
+        return self.subscriber_list
+
+    def get_number_subscribed_topics(self):
+
+        """
+        Return number of subscribed topics
+        """
+
+        return self.subscribe()
+
+    def get_number_failed_topics(self):
+
+        """
+        Return number of failed topics for subscription
+        """
+
+        return len(self.subscriber_list) - self.i
+
+    def get_topic_message_count(self, topic):
+
+        """
+        Return number of messages available in a topic
+        """
+
+        return len(self.topic_messages[topic])
+
+    def get_total_message_count(self):
+
+        """
+        Returns total number of messages across all topics
+        """
+
+        self.total_message_count = 0
+        for topic in self.topic_messages.keys():
+            self.total_message_count = self.total_message_count + self.get_topic_message_count(topic)
+
+        return self.total_message_count
+
+    def set_bag_directory(self):
+
+        """
+        Create ros bag save directory
+        """
+
+        rospack = rospkg.RosPack()
+        self.directory = os.path.join(rospack.get_path(self.dir), 'bags/')
+
+        rospy.loginfo('bag directory: %s', self.directory)
+        rospy.loginfo('bag name: %s', self.bag_name)
+
+        if not os.path.exists(self.directory):
+            os.makedirs(self.directory)
+
+        self.bag = rosbag.Bag(os.path.join(self.directory, self.bag_name + '.bag'), 'w')
+
+    def get_time_index(self, topic, requested_seconds):
+
+        """
+        Return the index for the time index for a topic at 'n' seconds from the end of the dequeue
+
+        For example, to bag the last 10 seconds of data, the index for 10 seconds back from the most
+        recent message can be obtained with this function.
+
+        The number of requested seconds should be the number of seoncds desired from
+        the end of deque. (ie. requested_seconds = 10 )
+
+        If the desired time length of the bag is greater than the available messages it will output a
+        message and return how ever many seconds of data are avaiable at the moment.
+
+        Seconds is of a number type (not a rospy.Time type) (ie. int, float)
+        """
+
+        topic_duration = self.get_topic_duration(topic).to_sec()
+
+        ratio = requested_seconds / topic_duration
+        index = math.floor(self.get_topic_message_count(topic) * (1 - min(ratio, 1)))
+
+        self.bag_report = 'The requested %s seconds were bagged' % requested_seconds
+
+        if not index:
+            self.bag_report = 'Only %s seconds of the request %s seconds were available, all \
+            messages were bagged' %  (topic_duration, requested_seconds)
+        return int(index)
+
+    def check_action(self, req):
```

Comment:
I think you should just accept a single unit, whether minutes or seconds. That way you can remove the unit field for the msg and it will be quicker to call, and less error prone.